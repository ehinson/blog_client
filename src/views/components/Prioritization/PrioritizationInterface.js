import React, { useState } from "react";
import { array, func, string } from "prop-types";
import _ from "lodash";

const sortAttributes = {
  created: "created_at",
  updated: "updated_at",
  pushed: "pushed_at",
  full_name: "full_name"
};

const propTypes = {
  repos: array.isRequired,
  setSortParams: func.isRequired,
  sort: string.isRequired,
  direction: string.isRequired
};

const PrioritizationInterface = ({ repos, setSortParams, sort, direction }) => {
  const [activeRepo, setActiveRepo] = useState({});
  return (
    <>
      <h2>Prioritization Interface</h2>
      <div
        onClick={() => setSortParams({ data: { key: "created" } })}
        isActive={sort === "created"}
      >
        created(default)
      </div>
      <div
        onClick={() => setSortParams({ data: { key: "updated" } })}
        isActive={sort === "updated"}
      >
        updated
      </div>
      <div
        onClick={() => setSortParams({ data: { key: "pushed" } })}
        isActive={sort === "pushed"}
      >
        pushed
      </div>
      <div
        onClick={() => setSortParams({ data: { key: "full_name" } })}
        isActive={sort === "full_name"}
      >
        full_name
      </div>
      <div
        onClick={() => setSortParams({ data: { order: "asc" } })}
        isActive={direction === "asc"}
      >
        asc
      </div>
      <div
        onClick={() => setSortParams({ data: { order: "desc" } })}
        isActive={direction === "desc"}
      >
        desc
      </div>
      <hr />
      <div>ActiveRepo</div>
      <div>{JSON.stringify(activeRepo)}</div>
      {_.orderBy(repos, [data => data[sortAttributes[sort]]], [direction]).map(
        repo => (
          <div
            key={repo.id}
            onClick={() =>
              setActiveRepo({ ownerID: repo.owner.id, repoID: repo.id })
            }
          >
            {repo.name}
          </div>
        )
      )}
    </>
  );
};

PrioritizationInterface.propTypes = propTypes;

export default PrioritizationInterface;
