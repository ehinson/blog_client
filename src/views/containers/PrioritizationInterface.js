import { compose } from 'recompose';
import { connect } from 'react-redux';

import { getUserRepos, getRepoIssues } from '../../state/selectors';

import PrioritizationInterface from '../components/Prioritization/PrioritizationInterface';
import { reduxForm } from 'redux-form';

const mapStateToProps = state => ({
  repos: getUserRepos(state),
  issues: getRepoIssues(state),
});

const mapDispatchToProps = dispatch => ({});

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  reduxForm({
    form: 'prioritization',
    initialValues: {
      sort: 'created',
      direction: 'desc',
    },
  }),
)(PrioritizationInterface);
