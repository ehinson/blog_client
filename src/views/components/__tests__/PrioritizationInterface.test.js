import React from 'react';
import { shallow } from 'enzyme';

// Components
import PrioritizationInterface from '../Prioritization/PrioritizationInterface';

const props = {
  hasToken: true,
  repos: [
    {
      id: 1234567890,
      name: 'My_cool_repo',
      owner: {
        login: 'ehinson',
      },
    },
  ],
  issues: [
    {
      assignee: null,
      created_at: '2019-11-12T20:14:36Z',
      updated_at: '2019-11-12T20:14:36Z',
    },
  ],
  fetchIssues: jest.fn(),
  isIssueLoading: false,
  setActiveRepo: jest.fn(),
  resetActiveRepo: jest.fn(),
  activeRepo: {},
};

describe('PrioritizationInterface Test Suite', () => {
  it('renders', () => {
    const wrapper = shallow(<PrioritizationInterface {...props} />);
    expect(wrapper.exists()).toBe(true);
  });
  it('Should match snapshot', () => {
    const wrapper = shallow(<PrioritizationInterface {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Should list repos', () => {
    const wrapper = shallow(<PrioritizationInterface {...props} />);
    console.log(wrapper.debug());
    expect(wrapper.find('PrioritizationInterface__StyledRepo').exists()).toBe(true);
  });
});
