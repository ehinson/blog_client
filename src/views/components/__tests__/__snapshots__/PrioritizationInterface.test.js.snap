// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrioritizationInterface Test Suite Should match snapshot 1`] = `
<Fragment>
  <PrioritizationInterface__StyledHeader
    role="banner"
  >
    <h1>
      Github Prioritization Interface
    </h1>
  </PrioritizationInterface__StyledHeader>
  <PrioritizationInterface__StyledWrapper
    isIssueLoading={false}
    role="main"
  >
    <PrioritizationInterface__StyledRepoWrapper
      tabIndex="0"
    >
      <PrioritizationInterface__StyledRepo
        active={false}
        key="1234567890"
        onClick={[Function]}
        tabIndex="0"
      >
        <div>
          <span>
            My_cool_repo
          </span>
        </div>
      </PrioritizationInterface__StyledRepo>
    </PrioritizationInterface__StyledRepoWrapper>
  </PrioritizationInterface__StyledWrapper>
  <PrioritizationInterface__StyledFooter />
</Fragment>
`;
